medcat/utils/normalizers.py

```diff
@@ -111,14 +101,93 @@ def edits2(self, word: str) -> Iterator[str]:
         Returns:
             Iterator[str]: All 2-away edits.
         """
-        return (e2 for e1 in self.edits1(word) for e2 in self.edits1(e1))
+        return get_all_edits_2(word, self.config.general.diacritics)
 
     def edits3(self, word):
         """All edits that are two edits away from `word`."""  # noqa
         # Do d3 edits
         pass
 
 
+def get_all_edits(word: str, use_diacritics: bool) -> Set[str]:
+    """Gets all the 1-distances edits of a word.
+
+    Args:
+        word (str): The word in question.
+        use_diacritics (bool): Whether or not to use diacritics.
+
+    Returns:
+        Set[str]: All the possibile edits.
+    """
+    letters    = 'abcdefghijklmnopqrstuvwxyz'
+
+    if use_diacritics:
+        letters += 'àáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ'
+
+    splits     = [(word[:i], word[i:])    for i in range(len(word) + 1)]
+    deletes    = [L + R[1:]               for L, R in splits if R]
+    transposes = [L + R[1] + R[0] + R[2:] for L, R in splits if len(R)>1]
+    replaces   = [L + c + R[1:]           for L, R in splits if R for c in letters]
+    inserts    = [L + c + R               for L, R in splits for c in letters]
+    return set(deletes + transposes + replaces + inserts)
+
+
+def get_all_edits_2(word: str, use_diacritics: bool,
+                    return_ordered: bool = False) -> Iterator[str]:
+    """Gets all the 2-distance edits of a word.
+
+    Args:
+        word (str): The word in question.
+        use_diacritics (bool): Whether or not to use diacritics
+        return_ordered (bool): Whether or not to use an ordered list of edits
+
+    Returns:
+        Iterator[Str]: Generator of all the possible words.
+    """
+    raw_fo_edits = get_all_edits(word, use_diacritics)
+    first_order_edits = sorted(raw_fo_edits) if return_ordered else raw_fo_edits
+    return (e2 for e1 in first_order_edits
+            for e2 in (sorted(get_all_edits(e1, use_diacritics))
+                       if return_ordered else
+                        get_all_edits(e1, use_diacritics)))
```

Comment:
Could the "second edit" in the inner loop modify a letter in the same position as the "first edit" already modified, essentially resulting in a 1-distance edit instead of 2?

Reply:
Kk, didn't notice it was there for a long time. So this getting "maximum 2-distance" edits method is highly likely not used anywhere or by anyone.